File: 백준/Silver/18311. 왕복/왕복.kt
package jimin.`11week`

import java.io.BufferedReader
import java.io.InputStreamReader

fun main(): Unit = with(BufferedReader(InputStreamReader(System.`in`))) {
    val (n, k) = readLine().trim().split(" ").map{ it.toLong() }
    val course = readLine().trim().split(" ").map{ it.toLong() }

    var distance = 0L
    kotlin.run {
        (course + course.reversed()).forEachIndexed { idx, it ->
            if (k in distance until  distance + it){
                println(
                    if (idx < n) idx + 1
                    else 2*n - idx
                )
                return@run
            }
            distance += it
        }
    }

}

File: 백준/Silver/18311. 왕복/왕복.py
n, k = map(int, input().split())
courses = list(map(int, input().split()))
now = 0
for idx, c in enumerate(courses + list(reversed(courses))):
    now += c
    if now > k:
        if idx < n:
            print(idx + 1)
            break
        else:
            print(2 * n - idx)
            break
    elif now == k:
        if idx < n:
            print(idx + 2)
            break
        else:
            print(2 * n - idx - 1)
            break
